src/client/js/Dialogs/Merge/MergeDialog.js

```diff
@@ -175,42 +172,62 @@ define([
             for (i = 0; i < mergeResult.conflict.items.length; i += 1) {
                 conflictItem = mergeResult.conflict.items[i];
                 conflictItemE = conflictItemTemplate.clone();
-                conflictItemE.find('.path').text(conflictItem.theirs.path);
 
-                linkTheirs = '?project=' + encodeURIComponent(self._client.getActiveProjectId()) +
-                    '&commit=' + encodeURIComponent(mergeResult.theirCommitHash) +
-                    // FIXME: regexp parses out the path
-                    '&node=' + encodeURIComponent(getParentPath(pathRegExp.exec(conflictItem.theirs.path)[0])) +
-                    '&selection=' + encodeURIComponent(pathRegExp.exec(conflictItem.theirs.path)[0]);
+                pathObject = DIFF.pathToObject(conflictItem.theirs.path);
+                conflictItemE.find('.path').text('"' + pathObject.node + '" [' +
+                    pathObject.full.substr(pathObject.node.length) + ']');
 
-                valueTheirsE = $('<div>' +
-                    '<a class="glyphicon glyphicon-link" href="' + linkTheirs + '" target="_blank" tooltip="Open"></a>' +
-                    '<span><xmp style="word-wrap: break-word; white-space: normal">' + JSON.stringify(conflictItem.theirs.value) + '</xmp></span>' +
+                link = '?project=' + encodeURIComponent(self._client.getActiveProjectId()) +
+                    '&commit=' + encodeURIComponent(mergeResult.theirCommitHash) +
+                    '&node=' + encodeURIComponent(getParentPath(pathObject.node)) +
+                    '&selection=' + encodeURIComponent(pathObject.node);
+
+                value = $('<div>' +
+                    '<a class="glyphicon glyphicon-link" href="' + link +
+                    '" target="_blank" tooltip="Open"></a>' +
+                    '<span><code style="word-wrap: break-word; white-space: normal">' +
+                    JSON.stringify(conflictItem.theirs.value) + '</code></span>' +
```

Comment:
String concatenation of dom elements are potentially bad, esp. for user input. If someone sets the value to `<script>alert("hello there!")</script>`. (I do see that it was like that before too..)